storage/src/mmr/mem.rs

```diff
@@ -899,23 +907,32 @@ mod tests {
             let root = mmr.root(&mut hasher);
 
             // Change a handful of leaves using a batch update.
+            let mut updates = Vec::new();
             for leaf in [0usize, 1, 10, 50, 100, 150, 197, 198] {
-                mmr.update_leaf_batched(&mut hasher, leaves[leaf], &element)
-                    .await
-                    .unwrap();
+                updates.push((leaves[leaf], &element));
             }
+            mmr.update_leaf_batched(&mut hasher, updates.into_iter())
+                .await
+                .unwrap();
+
             mmr.sync(&mut hasher);
             let updated_root = mmr.root(&mut hasher);
-            assert_ne!(root, updated_root);
+            assert_eq!(
+                "af3acad6aad59c1a880de643b1200a0962a95d06c087ebf677f29eb93fc359a4",
+                hex(&updated_root)
+            );
 
             // Batch-restore the changed leaves to their original values.
+            let mut updates = Vec::new();
             for leaf in [0usize, 1, 10, 50, 100, 150, 197, 198] {
                 c_hasher.update(&leaf.to_be_bytes());
                 let element = c_hasher.finalize();
-                mmr.update_leaf(&mut hasher, leaves[leaf], &element)
```

Comment:
Confirming this change from `update_leaf` to `update_leaf_batched` was intentional?

Reply:
yes, this test is for specifically testing batching.

Reply:
Got it, so this was an incorrect test 👍